src/Toml/Parser/Validate.hs

```diff
@@ -161,6 +162,7 @@ validateItemForest = go mempty
 
         TableName key -> do
             HashMap.lookup key tomlPairs `errorOnJust` SameNameKeyTable key
+            HashMap.lookup key tomlTableArrays `errorOnJust` SameNameTableArray key
```

Comment:
Wow, these tests are super useful! They already helped to find bugs! :exploding_head: